src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs

```diff
@@ -69,6 +139,83 @@ interface ICLRRuntimeInfo
             // Don't need any other methods.
         }
 
+#if NET6_0_OR_GREATER
+        private class CLRRuntimeInfoWrapperCache : ComWrappers
+        {
+            public static readonly CLRRuntimeInfoWrapperCache Instance = new CLRRuntimeInfoWrapperCache();
+            private CLRRuntimeInfoWrapperCache() { }
+
+            protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count) => throw new NotImplementedException();
+            protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+            {
+                Debug.Assert(flags == CreateObjectFlags.UniqueInstance);
+
+                return new ClrRuntimeInfoRcw(externalComObject);
+            }
+            protected override void ReleaseObjects(IEnumerable objects) => throw new NotImplementedException();
+
+            private unsafe sealed record ClrRuntimeInfoRcw(IntPtr Inst) : ICLRRuntimeInfo, IDisposable
```

Comment:
Nit/rant: I really wish C# just had primary constructors instead of records - it feels like in places like this, the 7 additional methods, 2 operators, and 1 property that get inflicted by typing `record` instead of `class` are quite unjustified compared to the useful parts we get out of it (a syntax shortcut for not having to declare+initialize a field). Most of this overhead is also not possible to trim because it's referenced from ToString/Equals.

If we were to start using records heavily, it would probably accumulate to a measurable/meaningful size difference.

Reply:
I explicitly designed primary constructors at the same time as I did records, so it could be added. I don't think Roslyn has time to add it yet, but it's on the roadmap AFAIK. Syntax proposed is just

```C#
class MyClass(int _privateField) { ... }
```

Leaving space for primary constructors is why records use a different keyword.